assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -44,12 +48,18 @@ interactivityApiStore( {
 					context?.woocommerce.imageId
 				);
 			},
+			isDialogOpen: ( { context }: Store ) => {
+				return context?.woocommerce.isDialogOpen;
+			},
 		},
 	},
 	actions: {
 		woocommerce: {
-			handleClick: ( { context }: Store ) => {
-				context.woocommerce.selectedImage = context.woocommerce.imageId;
+			thumbnails: {
+				handleClick: ( { context }: Store ) => {
+					context.woocommerce.selectedImage =
+						context.woocommerce.imageId;
```

Comment:
@danieldudzic You were right about https://github.com/woocommerce/woocommerce-blocks/pull/10776#discussion_r1321319877.

It seems that the actions are different from the selectors 🤔. My suggestion is to try to keep the structure flatten and nested only when it is necessary. What do you think?

Reply:
I'm warm towards both of the mentioned approaches; however, I think it would be useful to come up with a convention that we could follow for all Interactivity API implementations across different blocks.

I don't see a major drawback to using the nested structure - at least then it will always be predictable.